### DISCUSSION TITLE
Build Error: GenerateFileTask Instantiation Failure

### DISCUSSION THREAD

**Discussion by WenHong-Lau** [unoplatform/uno#18980] [General]
I'm encountering a build error in my project that appears to be related to Windows updates. The issue has been reproduced on two separate machines, both occurring after Windows updates were installed.

`
The "GenerateFileTask" task could not be instantiated from "C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\1.4.0\build\../tools/Uno.Dsp.Tasks.dll". 
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.Extensions.DependencyInjection.Abstractions, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
   at Microsoft.Extensions.DependencyInjection.LoggingServiceCollectionExtensions.AddLogging(IServiceCollection services, Action`1 configure)
   at Microsoft.Extensions.Hosting.HostBuilder.PopulateServiceCollection(IServiceCollection services, HostBuilderContext hostBuilderContext, HostingEnvironment hostingEnvironment, PhysicalFileProvider defaultFileProvider, IConfiguration appConfiguration, Func`1 serviceProviderGetter)
   at Microsoft.Extensions.Hosting.HostBuilder.InitializeServiceProvider()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at Uno.Dsp.Tasks.TaskBase..ctor()

WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
	Datgel.Acquisition (net8.0)	C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\1.4.0\build\Uno.Dsp.Tasks.targets	20	
`

**Comment by WenHong-Lau** [comment]
Might not be Windows update, not sure what is causing this

**Reply by jeromelaban** [reply]
Thanks for the report, let's track in https://github.com/unoplatform/uno/issues/19012.

**Comment by WenHong-Lau** [comment]
PS C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\1.4.0\tools> ls
Directory: C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\1.4.0\tools

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         15/6/2024   2:53 am          28384 Microsoft.Bcl.AsyncInterfaces.dll
-a----         15/6/2024   2:53 am          32480 Microsoft.Extensions.Configuration.Abstractions.dll
-a----         15/6/2024   2:53 am          49888 Microsoft.Extensions.Configuration.Binder.dll
-a----         15/6/2024   2:53 am          29920 Microsoft.Extensions.Configuration.CommandLine.dll
-a----         15/6/2024   2:53 am          48864 Microsoft.Extensions.Configuration.dll
-a----         15/6/2024   2:53 am          26336 Microsoft.Extensions.Configuration.EnvironmentVariables.dll
-a----         15/6/2024   2:53 am          32992 Microsoft.Extensions.Configuration.FileExtensions.dll
-a----         15/6/2024   2:53 am          31968 Microsoft.Extensions.Configuration.Json.dll
-a----         15/6/2024   2:53 am          30432 Microsoft.Extensions.Configuration.UserSecrets.dll
-a----         15/6/2024   2:53 am          65760 Microsoft.Extensions.DependencyInjection.Abstractions.dll
-a----         15/6/2024   2:53 am          92384 Microsoft.Extensions.DependencyInjection.dll
-a----         15/6/2024   2:53 am          36064 Microsoft.Extensions.Diagnostics.Abstractions.dll
-a----         15/6/2024   2:53 am          40160 Microsoft.Extensions.Diagnostics.dll
-a----         15/6/2024   2:53 am          27360 Microsoft.Extensions.FileProviders.Abstractions.dll
-a----         15/6/2024   2:53 am          49888 Microsoft.Extensions.FileProviders.Physical.dll
-a----         15/6/2024   2:53 am          50912 Microsoft.Extensions.FileSystemGlobbing.dll
-a----         15/6/2024   2:53 am          57056 Microsoft.Extensions.Hosting.Abstractions.dll
-a----         15/6/2024   2:53 am          80608 Microsoft.Extensions.Hosting.dll
-a----         15/6/2024   2:53 am          69344 Microsoft.Extensions.Logging.Abstractions.dll
-a----         15/6/2024   2:53 am          33504 Microsoft.Extensions.Logging.Configuration.dll
-a----         15/6/2024   2:53 am          80096 Microsoft.Extensions.Logging.Console.dll
-a----         15/6/2024   2:53 am          25312 Microsoft.Extensions.Logging.Debug.dll
-a----         15/6/2024   2:53 am          55520 Microsoft.Extensions.Logging.dll
-a----         15/6/2024   2:53 am          31968 Microsoft.Extensions.Logging.EventLog.dll
-a----         15/6/2024   2:53 am          41696 Microsoft.Extensions.Logging.EventSource.dll
-a----         15/6/2024   2:53 am          29408 Microsoft.Extensions.Options.ConfigurationExtensions.dll
-a----         15/6/2024   2:53 am          70368 Microsoft.Extensions.Options.dll
-a----         15/6/2024   2:53 am          51424 Microsoft.Extensions.Primitives.dll
-a----         15/6/2024   2:53 am         371424 Mono.Cecil.dll
-a----         15/6/2024   2:53 am          51424 Mono.Cecil.Mdb.dll
-a----         15/6/2024   2:53 am         101600 Mono.Cecil.Pdb.dll
-a----         15/6/2024   2:53 am          38624 Mono.Cecil.Rocks.dll
-a----         15/6/2024   2:53 am          24288 System.Buffers.dll
-a----         15/6/2024   2:53 am         254176 System.Collections.Immutable.dll
-a----         15/6/2024   2:53 am          77024 System.ComponentModel.Annotations.dll
-a----         15/6/2024   2:53 am         176864 System.Diagnostics.DiagnosticSource.dll
-a----         15/6/2024   2:53 am          57056 System.Diagnostics.EventLog.dll
-a----         15/6/2024   2:53 am          85728 System.IO.Pipelines.dll
-a----         15/6/2024   2:53 am         144608 System.Memory.dll
-a----         15/6/2024   2:53 am         159456 System.Numerics.Vectors.dll
-a----         15/6/2024   2:53 am          20704 System.Runtime.CompilerServices.Unsafe.dll
-a----         15/6/2024   2:53 am          40672 System.Security.Principal.Windows.dll
-a----         15/6/2024   2:53 am          80608 System.Text.Encodings.Web.dll
-a----         15/6/2024   2:53 am         608992 System.Text.Json.dll
-a----         15/6/2024   2:53 am          28896 System.Threading.Tasks.Extensions.dll
-a----         15/6/2024   2:53 am          74464 Uno.Core.Extensions.dll
-a----         15/6/2024   2:53 am          73952 Uno.Dsp.dll
-a----         15/6/2024   2:53 am          33504 Uno.Dsp.Tasks.dll
-a----         15/6/2024   2:53 am         115424 Uno.Extensions.XamlToCsharp.dll


PS C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\1.4.0\tools> sn -T .\Microsoft.Extensions.DependencyInjection.Abstractions.dll

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Public key token is adb9793829ddae60

**Comment by WenHong-Lau** [comment]
[logs.zip](https://github.com/user-attachments/files/17974926/logs.zip)

**Comment by ajpinedam** [comment]
Hi, @WenHong-Lau. We moved the discussion to ticket # https://github.com/unoplatform/uno/issues/18986, but I was trying to tag you from there, and for some reason, GitHub did not allow me.

I am looking into this issue, but I have been unable to reproduce it. I have a couple of questions:

What kind of project are you running (Desktop, mobile, WASM, etc.)?
Verion of Uno.SDK you are using?
Current Version of Windows?
Version of Visual Studio?
.net8.0 or .net9.0?

If you create a new Uno Platform project, do you get the same error?

Finally, could you delete the directory  `C:\Users\wenhong.lau.DATGEL\.nuget\packages\uno.dsp.tasks\` and everything inside it and try rebuilding the project?

I created a new app and upgraded my Windows PC to the latest, currently running on Windows 11 Version `10.0.26100 Build 26100`

I created two projects, one using the template in Visual Studio and selected and another from the command line utility. Both projects were: 

net9.0,  MVUX, XAML, Material, Extension, etc., most options are chosen.

Please reply directly to the issue https://github.com/unoplatform/uno/issues/18986 so we can continue the conversation there.
